server/store/memory/peer_store.go

```diff
@@ -127,16 +132,15 @@ func (s *peerStore) DeleteSeeder(infoHash chihaya.InfoHash, p chihaya.Peer) erro
 
 func (s *peerStore) PutLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error {
 	key := leechersKey(infoHash)
+	i := s.shardIndex(infoHash)
```

Comment:
you can go back to using `shard := s.shards[s.shardIndex(infoHash)]`, like you did for `GetSeeders`